core/src/main/java/io/grpc/ClientCall.java

```diff
@@ -157,15 +159,24 @@ public void onReady() {}
   public abstract void request(int numMessages);
 
   /**
+   * Equivalent as {@code cancel(null, null)}.
+   */
+  public final void cancel() {
+    cancel(null, null);
+  }
+
+  /**
    * Prevent any further processing for this {@code ClientCall}. No further messages may be sent or
    * will be received. The server is informed of cancellations, but may not stop processing the
    * call. Cancellation is permitted if previously {@link #halfClose}d. Cancelling an already {@code
    * cancel()}ed {@code ClientCall} has no effect.
    *
-   *
    * <p>No other methods on this class can be called after this method has been called.
+   *
+   * @param message if not {@code null}, will appear as the description of the CANCELLED status
+   * @param cause if not {@code null}, will appear as the cause of the CANCELLED status
    */
-  public abstract void cancel();
+  public abstract void cancel(@Nullable String message, @Nullable Throwable cause);
```

Comment:
They will get a CANCELLED status, whose cause is a non-CANCELLED StatusException, which means "I cancelled this call due to the given non-CANCELLED status", which makes sense to me.